modules/ducktests/tests/ignitetest/services/utils/ignite_aware.py

```diff
@@ -73,16 +82,31 @@ def __init__(self, context, config, num_nodes, startup_timeout_sec, shutdown_tim
         self.main_java_class = main_java_class
         self.startup_timeout_sec = startup_timeout_sec
         self.shutdown_timeout_sec = shutdown_timeout_sec
+        self.thin_client_config = thin_client_config
 
-        self.spec = resolve_spec(self, context, config, main_java_class, **kwargs)
+        self.spec = resolve_spec(self, context, config, main_java_class,
+                                 self.mode, thin_client_config, **kwargs)
         self.init_logs_attribute()
 
         self.disconnected_nodes = []
-        self.start_ignite = kwargs.get("start_ignite", True)
 
+    # pylint: disable=R1705
     @property
     def version(self):
-        return self.config.version
+        if not self.thin_client_config:
+            return self.config.version
+        else:
+            return self.thin_client_config.version
+
+    @property
+    def mode(self):
+        if self.thin_client_config:
+            return IgniteAwareService.StartIgnite.THIN_CLIENT
+        else:
+            if self.config:
```

Comment:
This can be rewritten as

`return IgniteAwareService.StartIgnite.NODE if self.config else IgniteAwareService.StartIgnite.NONE`

Please, import `StartIgnite` to use shorten names.